pd_xpyd/readme.md

```diff
@@ -0,0 +1,167 @@
+# Guide to Setting Up PD Disaggregation with Mooncake
+
+> **Note**: This document does not cover MLA data parallel setup. Currently, PD disaggregation is only supported with vllm-hpu v0.
+
+## Mooncake Installation
+
+### Install via pip
+
+```bash
+apt install etcd -y
+pip install mooncake-transfer-engine==0.3.0b3
+```
+
+### Install from Source
+
+```bash
+# Install required packages
+apt install git wget curl net-tools sudo iputils-ping etcd -y
+
+# Clone the Mooncake repository
+git clone https://github.com/kvcache-ai/Mooncake.git -b v0.3.0-beta
+cd Mooncake
+
+# Install dependencies
+bash dependencies.sh
+
+# Build and install Mooncake
+mkdir build
+cd build
+cmake ..
+make -j
+make install
+```
+
+### Setting up RDMA (Optional)
+
+To enable RDMA for high-speed data transfer, follow these steps:
+
+```bash
+apt remove ibutils libpmix-aws
+wget https://www.mellanox.com/downloads/DOCA/DOCA_v2.10.0/host/doca-host_2.10.0-093000-25.01-ubuntu2204_amd64.deb
+dpkg -i doca-host_2.10.0-093000-25.01-ubuntu2204_amd64.deb
+apt-get update
+apt-get -y install doca-ofed
+
+# Check RDMA devices and network interfaces
+ibdev2netdev
+# Example output:
+# mlx5_0 port 1 ==> ens108np0 (Up)
+# mlx5_1 port 1 ==> ens9f0np0 (Up)
+# ...
+```
+
+> **Note**: If you are running RDMA inside a Docker container, add the `--privileged` flag to the `docker run` command to ensure proper hardware and network access.
```

Comment:
Please also mention enabling Transparent Huge Page here. Otherwise, we'll see decoding degradation compared to TCP.